libkbfs/folder_branch_ops.go

```diff
@@ -7424,6 +7433,173 @@ func (fbo *folderBranchOps) Reset(
+		}
+	} else {
+		block = NewFileBlock().(*FileBlock)
+		err = assembleBlock(
+			ctx, fbo.config.keyGetter(), fbo.config.Codec(),
+			fbo.config.Crypto(), kmd, config.Paths.Ptr, block,
+			config.Paths.Buf, config.Paths.ServerHalf)
+		if err != nil {
+			return keybase1.FolderSyncConfig{}, err
+		}
+	}
+
+	paths, err := syncPathListFromBlock(fbo.config.Codec(), block)
+	if err != nil {
+		return keybase1.FolderSyncConfig{}, err
+	}
+	ret.Paths = paths.Paths
+	return ret, nil
+}
+
+// GetSyncConfig implements the KBFSOps interface for folderBranchOps.
+func (fbo *folderBranchOps) GetSyncConfig(
+	ctx context.Context, tlfID tlf.ID) (keybase1.FolderSyncConfig, error) {
+	if tlfID != fbo.id() || fbo.branch() != MasterBranch {
+		return keybase1.FolderSyncConfig{}, WrongOpsError{
+			fbo.folderBranch, FolderBranch{tlfID, MasterBranch}}
+	}
+
+	lState := makeFBOLockState()
+	md, _ := fbo.getHead(ctx, lState, mdNoCommit)
+
+	fbo.syncLock.RLock(lState)
+	defer fbo.syncLock.RUnlock(lState)
+	return fbo.getProtocolSyncConfig(ctx, lState, md)
+}
+
+func (fbo *folderBranchOps) makeEncryptedPartialPathsLocked(
+	ctx context.Context, lState *lockState, kmd KeyMetadata, paths []string) (
+	FolderSyncEncryptedPartialPaths, error) {
+	fbo.syncLock.AssertLocked(lState)
+
+	oldConfig, err := fbo.getProtocolSyncConfig(ctx, lState, kmd)
+	if err != nil {
+		return FolderSyncEncryptedPartialPaths{}, err
+	}
+	if oldConfig.Mode == keybase1.FolderSyncMode_ENABLED {
+		return FolderSyncEncryptedPartialPaths{},
+			errors.Errorf("TLF %s is already fully synced", fbo.id())
+	}
+
+	// Make sure the new path list doesn't contain duplicates,
+	// contains no duplicates, and each path is cleaned.
+	seenPaths := make(map[string]bool, len(paths))
+	var pathList syncPathList
+	pathList.Paths = make([]string, len(paths))
+	for i, p := range paths {
+		p = stdpath.Clean(filepath.ToSlash(p))
+		if seenPaths[p] {
+			return FolderSyncEncryptedPartialPaths{}, errors.Errorf(
+				"%s is in the paths list more than once", p)
+		}
+		if stdpath.IsAbs(p) {
+			return FolderSyncEncryptedPartialPaths{}, errors.Errorf(
+				"Absolute paths like %s are not allowed", p)
+		}
+		if strings.HasPrefix(p, "..") {
+			return FolderSyncEncryptedPartialPaths{}, errors.Errorf(
+				"Relative paths out of the TLF like %s are not allowed", p)
+		}
+		seenPaths[p] = true
+		pathList.Paths[i] = p
+	}
+
+	fbo.log.CDebugf(ctx,
+		"Setting partial sync config for %s; paths=%v",
+		fbo.id(), pathList.Paths)
+
+	b, err := pathList.makeBlock(fbo.config.Codec())
+	if err != nil {
+		return FolderSyncEncryptedPartialPaths{}, err
+	}
+
+	chargedTo, err := chargedToForTLF(
+		ctx, fbo.config.KBPKI(), fbo.config.KBPKI(), kmd.GetTlfHandle())
+	if err != nil {
+		return FolderSyncEncryptedPartialPaths{}, err
+	}
+
+	info, _, readyBlockData, err :=
+		ReadyBlock(ctx, fbo.config.BlockCache(), fbo.config.BlockOps(),
+			fbo.config.Crypto(), kmd, b, chargedTo,
+			fbo.config.DefaultBlockType())
```

Comment:
So since we don't have indirect blocks for path storage, we implicitly have a cap for the number of partial paths that we can sync. IMO it's high enough that we can pretty much ignore this, but it might be good to call it out somewhere.

Also worth calling out that the block is never synced remotely--that the partial sync state is per-client only.

Reply:
> So since we don't have indirect blocks for path storage, we implicitly have a cap for the number of partial paths that we can sync. IMO it's high enough that we can pretty much ignore this, but it might be good to call it out somewhere.

Since we're not sending these blocks to the server, there's really no limit on the size of the block, except the user's memory.  Is that what you meant?   Not sure if that's worth mentioning or not, since anything that hits the user's memory limit will probably just crash things before it's even committed to disk.